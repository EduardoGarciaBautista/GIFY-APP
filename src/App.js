import React, { useState } from "react";
import AddCategory from "./Components/AddCategory";
import  GifGrid  from "./Components/GifGrid";


export const GifExpertApp = ({defaultCategories = []}) => {


    const [categories, setcategories] = useState(defaultCategories);

    return (
        <>
            <h2>Gif Expert</h2>
            <AddCategory setcategories={setcategories} />
            <hr />

            <ol>
                {categories && categories.map(item => {
                    return (
                        <GifGrid
                            key={item}
                            category={item}
                        />
                    )
                })
                }
            </ol>

        </>
    )
}
