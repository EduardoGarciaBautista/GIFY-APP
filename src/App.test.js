import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifExpertApp } from './App';


describe('Tests for App', () => {
    

    test('should Match with snapshot', () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should show a categories list', () => {

        const categories= ['Goku'];

        const wrapper = shallow(<GifExpertApp 
            defaultCategories={categories}/>);

            expect(wrapper).toMatchSnapshot();

            expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
    


})