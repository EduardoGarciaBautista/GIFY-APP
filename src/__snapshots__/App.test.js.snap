// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for App should Match with snapshot 1`] = `
<Fragment>
  <h2>
    Gif Expert
  </h2>
  <AddCategory
    setcategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Tests for App should show a categories list 1`] = `
<Fragment>
  <h2>
    Gif Expert
  </h2>
  <AddCategory
    setcategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;
